docs/advanced/visual-studio-code.md

```diff
@@ -0,0 +1,91 @@
+
+## Setting Up a Worker Node
+
+Log into the cluster and use [`salloc`](https://slurm.schedmd.com/salloc.html) to request an interactive job with reasonable resources. Here we ask for 4 GB of memory, which should be enough for most development tasks, and 3 hours of runtime, which is the maximum allowed under the free tier. Take note of which node the job runs on.
+
+> Note: VS Code's remote development extension works over SSH. At the time of this writing, the only worker node you can SSH into is node17. Request node17 specifically using `salloc -w node17`
+
+```
+[localuser@localmachine ~]$ ssh login3.chpc.wustl.edu
+Last login: Tue Sep 19 14:15:38 2023 from 10.20.145.192
+[clusteruser@login02 ~]$ salloc -w node17 --nodes=1 --time=3:00:00 --mem=4GB --qos=interactive
+salloc: Nodes node17 are ready for job
+[clusteruser@node17 ~]$ 
+```
+
+You can also figure out which worker node your job is running on with [`squeue`](https://slurm.schedmd.com/squeue.html).
+
+```
+[clusteruser@login02 ~] squeue -u clusteruser -t running
+JOBID      PARTITION    NAME        USER           ST    TIME    NODES    NODELIST(REASON)
+4074089    small        interact    clusteruser    R     7:58    1        node17
+```
+
+And when you are finished working, you can cancel the job.
+
+```
+[clusteruser@login02 ~] scancel 4074089 # use job id from squeue
+```
+
+## Setting Up Visual Studio Code
+
+> Note: VS Code has a little bit of trouble with Slurm's module system. Append the following to your `~/.bashrc` on the cluster:
+>
+> ```
+> export MODULEPATH=$MODULEPATH:/opt/modulefiles
+> ```
+
+Follow instructions to install [VS Code's remote development over SSH extension](https://code.visualstudio.com/docs/remote/ssh). Pick a port on your local computer other than 22 to tunnel to VS Code's remote development server on the cluster. In this example we'll use local port 2222. To tell VS Code about the connection, edit `~/.ssh/config` on your local computer and add a host for the cluster. You can name the host whatever you want -- here we'll call it `chpc-vscode`.
+
+```
+Host chpc-vscode
+  Hostname 127.0.0.1            
+  User clusteruser
+  IdentityFile ~/.ssh/chpc_rsa
+  Port 2222
+```
+
+Note that you *must* set up public key authentication as described [here](../getting-started/connect-to-login-nodes.md). Password authentication is *not* allowed on the worker nodes. In this example, our private key is stored in `~/.ssh/chpc_rsa` on our local computer.
```

Comment:
nit: could you change the hyperlink to `../getting-started/connect-to-login-nodes.md#entering-my-password-to-login-every-time-is-so-annoying--how-can-i-connect-to-the-cluster-without-entering-the-password`, so users are sent to the exact section?

Reply:
Good call, added #id to that link and one other in `jupyter.md`.